tests/integration.rs

```diff
@@ -341,3 +372,89 @@ fn subcommand_usage() -> TestResult {
 fn standalone_usage() -> TestResult {
     golden_reference(&["-h"], "tests/standalone-usage.txt")
 }
+
+/// This test is a bit more verbose than the other tests because it uses a slightly
+/// different mechanism for setting up and sweeping / cleaning.
+///
+/// Step-by-step:
+///
+/// * Create a new temporary directory. Other tests only output the built files into
+///   the target directory, but this test copies the whole template project into the
+///   target directory.
+/// * Build the two different (nested) projects using cargo build, but without incremental
+///   compilation, because cargo-sweep doesn't remove files built by incremental compilation,
+///   which affects this test since this test doesn't use a separate target folder for just
+///   the target files.
+/// * Do a dry run sweep, asserting that nothing was actually cleaned.
+/// * Do a proper sweep, asserting that the target directory size is back down to its
+///   expected size.
+///
+/// Please note that there is some ceremony involved, namely correctly setting:
+/// * `CARGO_TARGET_DIR` to `unset`, so as not to clash with users who use `CARGO_TARGET_DIR`
+///   when they invoke `cargo test` for running these tests.
+/// * `CARGO_INCREMENTAL` to `0`, because cargo-sweep doesn't yet clear incremental files.
+#[test]
+fn recursive_multiple_root_workspaces() -> TestResult {
```

Comment:
This test is not effective, it passes even without your change.
```
; git log -n1
commit 8ca98385c1d7af79b491a30eb330f12a4b7173e2 (HEAD)
Author: Erik Zivkovic <erik@zivkovic.se>
Date:   Wed Feb 1 23:13:52 2023 +0100

    Make cargo-sweep output parselable by human-size
; git status 
HEAD detached at FETCH_HEAD
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   src/main.rs
        modified:   src/util.rs

; git diff origin/master src
; cargo test recursive -q

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 3 filtered out; finished in 0.00s


running 1 test
.
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 9 filtered out; finished in 1.03s
```

Reply:
I agree that the last change `8ca98385c1d7af79b491a30eb330f12a4b7173e2` doesn't do anything locally, but as you can see it failed CI because for some reason that windows machine generated a line that looks like this:

```
[INFO] Would clean: 431.00 bytes from "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\.tmpusn7fA\\nested-root-workspace\\bin-crate\\target"
```

See output of this test run
https://github.com/holmgr/cargo-sweep/actions/runs/4062073908/jobs/7008162492

Reply:
You're misunderstanding, sorry - I'm not saying 8ca98385c1d7af79b491a30eb330f12a4b7173e2 doesn't do anything, I'm saying "this test doesn't test anything". It passes both before and after all your changes to `src/`.

https://github.com/holmgr/cargo-sweep/actions/runs/4062073908/jobs/7008162492 looks like a difference in the size of the target directory, it makes sense to me that 8ca98385c1d7af79b491a30eb330f12a4b7173e2 fixes it.

Reply:
Gotcha, I understand now. Done.